File: src/game/tools/rpc/output/wrap/HeroListDataWrap.go
package wraps

import (
	rpc "tipcat.com/common/rpc/simple"
)

type HeroListDataWrap struct {
	
	HerosList []*HeroDataWrap // 英雄列表
	rpc.Wrapper
}

func (w *HeroListDataWrap)Decode(pck *rpc.Packet) rpc.Wrapper {
	
	w.HerosList = make([]*HeroDataWrap, int(pck.PopInt16()))
	for i := 0; i < len(w.HerosList); i++ {
		w.HerosList[i] = new(HeroDataWrap)
		w.HerosList[i].Decode(pck)
		
	}
	
	return w
}

func (w *HeroListDataWrap)Encode(pck *rpc.Packet) {
	
	if w.HerosList == nil {
		pck.PutInt16(0)
	} else {
    	pck.PutInt16(int16(len(w.HerosList)))
    	for i := 0; i < len(w.HerosList); i++ {
    		w.HerosList[i].Encode(pck);
			
    	}
    }
	
}

File: src/game/tools/rpc/output/wrap/HeroListDataWrap.cs
using System;
using Common.Net.Simple;

namespace Game 
{
	public class HeroListDataWrap : IWrapper
	{	
		public HeroDataWrap[] HerosList; // 英雄列表
		
		public void Decode(Packet pck)
		{	
			HerosList = new HeroDataWrap[pck.GetShort()];
			for (int i = 0; i < HerosList.Length; i++)
			{
				HerosList[i] = new HeroDataWrap();
				HerosList[i].Decode(pck);
				
			}
			
		}
		
        public void Encode(Packet pck)
        {	
        	if (HerosList == null) pck.PutShort((short)0); 
        	else
        	{
	        	pck.PutShort((short)HerosList.Length);
	        	for(int i = 0; i < HerosList.Length; i++)
	        	{
	        		HerosList[i].Encode(pck);
					
	        	}
	        }
        	
        }
	}
}